src/components/WalletSelector/WalletSelector.tsx

```diff
@@ -8,24 +8,54 @@ import { useAuthActions, useAuthState } from 'contexts/auth/AuthContext';
 import { useModal } from 'contexts/modal/ModalContext';
 import WalletButton from './WalletButton';
 import { isLoggedInState } from 'contexts/auth/types';
+import colors from 'components/core/colors';
 
 const walletConnectorMap: Record<string, AbstractConnector> = {
   Metamask: injected,
   WalletConnect: walletconnect,
 };
 
+type ErrorMessage = {
+  heading: string;
+  body: string;
+};
+
+const ERROR_MESSAGES: { [key: string]: ErrorMessage } = {
+  REJECTED_SIGNATURE: {
+    heading: 'Signature rejected',
+    body: 'Please sign the message with your wallet to continue',
+  },
+  UNKNOWN_ERROR: {
+    heading: 'There was an error connecting',
+    body: 'Please try again',
+  },
+};
+
+function getErrorMessage(errorCode: string) {
+  return ERROR_MESSAGES[errorCode]
+    ? ERROR_MESSAGES[errorCode]
+    : ERROR_MESSAGES.UNKNOWN_ERROR;
```

Comment:
here you could use something fun called [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator): `return ERROR_MESSAGES[errorCode] ?? ERROR_MESSAGES.UNKNOWN_ERROR`

Reply:
done ✅